app/static/js/view_map.js

```diff
@@ -7,22 +9,63 @@ $(document).ready(function () {
 });
 
 function grabMyPosition() {
-    if (navigator.geolocation) {
-        return navigator.geolocation.getCurrentPosition(centerMap);
+    var geoOptions = { maximumAge: 30000,  //  Valid for 5 minutes
+        timeout:5000,  // Wait 5 seconds
+        enableHighAccuracy:true
+    }

```

Comment:
This file also has much of the same logic as in report_map.js and trip_planner.js. All this identical code should be refactored so that it's just in one place. Note that this comment applies to much of the content in this file.